hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1195,8 +1198,7 @@ protected HoodieData<HoodieRecord> prepRecords(Map<MetadataPartitionType,
   public void performTableServices(Option<String> inFlightInstantTimestamp) {
     HoodieTimer metadataTableServicesTimer = HoodieTimer.start();
     boolean allTableServicesExecutedSuccessfullyOrSkipped = true;
-    try {
-      BaseHoodieWriteClient writeClient = getWriteClient();
+    try (BaseHoodieWriteClient<?, I, ?, ?> writeClient = getWriteClient()) {
```

Comment:
sorry, we might have to revert this. we try to re-use an existing instance of write client